edith-step/src/App.js

```diff
@@ -9,38 +9,48 @@ import axios from 'axios';
 class App extends Component {
   constructor() {
     super();
-    this.state = { "chartType" : LineChart }; 
-    this.updateChartType = this.updateChartType.bind(this);   
+    this.state = { "chartType" : LineChart, "dateSelection": "" }; 
+    this.updateChartType = this.updateChartType.bind(this); 
+    this.showWeeklyChart = this.showWeeklyChart.bind(this);  
+    this.revertChart = this.revertChart.bind(this);
   }
 
   updateChartType(event) {
     switch(event.target.value) {
-      case "LineChart": 
+      case "line": 
```

Comment:
It might be useful to us an enum here to be more explicit and rigid. lmk in case I missed anything.

pseudocode
```
enum {
 LINE,
 BAR,
 DOUGHNUT
}